src/main/java/core/basesyntax/strategy/impl/ReturnOperation.java

```
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+
+public class ReturnOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction) {
+        if (transaction == null) {
+            throw new RuntimeException("Transaction cannot be null.");
+        }
+        int currentQuantity = Storage.getFruitQuantity(transaction.getFruit());
```

You check `transaction` for null (good) but do not validate `transaction.getFruit()` or `transaction.getQuantity()`. The checklist requires anticipating invalid input and handling null parameters: "Remember that your code will need to be tested, so try to anticipate and handle all invalid input data in advance. For example: ... 3. Null parameters". Add validations such as `if (transaction.getFruit() == null || transaction.getFruit().trim().isEmpty()) { throw new RuntimeException("Fruit name cannot be null or empty"); }` and `if (transaction.getQuantity() < 0) { throw new RuntimeException("Quantity cannot be negative"); }` to fail fast with informative messages.